packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -440,11 +441,105 @@ class RenderParagraph extends RenderBox {
   @override
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
-    config
-      ..label = text.toPlainText()
-      ..textDirection = textDirection;
+    bool hasRecognizers = false;
+    text.visitTextSpan((TextSpan span) {
+      if (span.recognizer != null) {
+        hasRecognizers = true;
+        return false;
+      }
+      return true;
+    });
+    if (hasRecognizers) {
+      config.explicitChildNodes = true;
+      config.isSemanticBoundary = true;
+    } else {
+      config.label = text.toPlainText();
+      config.textDirection = textDirection;
+    }
   }
 
+  @override
+  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
+    final List<SemanticsNode> newChildren = <SemanticsNode>[];
```

Comment:
can you assert that this isn't entered if we don't have recognizers?

Reply:
Added several asserts